### ISSUE TITLE
Parsing is inconsistent with header regex

### ISSUE THREAD

**Issue by raphaelhuleux** [Pseudonium/Obsidian_to_Anki#73]
Hello, 

I am trying to import a markdown with a dozen cards. However, it doesn't recognize some of the last cards for some unknown reason. I am attaching the problematic file.

Also, as a potential feature request, do you think it could be possible to modify the regex so that the answers from a card can include blank lines? I tried a solution myself with the following regex but without success: ``^## (.+)\n+((?:.|\n)+?END)``

**Comment by ddfonseca** [comment]
I don't know why, but I put the FILE TAGS on the beginning of the file and the regex provided on the documentation: 
`**Regex line**: ^#+(.+)\n+((?:[^\n#][\n]?)+)`
Works fine, for the 14 cards.

For the blank line I use <br/> , like I said [here](https://github.com/Pseudonium/Obsidian_to_Anki/issues/67#issuecomment-698076202)

**Comment by raphaelhuleux** [comment]
Thank you it seems that it works better with the tag at the beginning of the file.
However, I also noticed a strange behavior with a personal regex that I use for vocabulary: ``^- (.*) \/ (.*)\n``
It seems that, when I add a card to a file and launch the script again, it creates doublons (double ids for existing cards). 

I also attached the problematic files.
[Archive.zip](https://github.com/Pseudonium/Obsidian_to_Anki/files/5336861/Archive.zip)

**Comment by ddfonseca** [comment]
Here works fine too. I ran the script more than 3 times and it doesn't create another ID's with your custom regex. I think you may have more than one note type with the same regex or conflicting's one. Try to make it consistent between note types and for the file. 

Test deleting the other custom regex and leave this, then try again. Because here works fine

**Comment by ddfonseca** [comment]
Actually, only the last one didn't update and I don't know why

**Comment by raphaelhuleux** [comment]
Hey, thank you for your answer. The script works fine if you don't change the file, but if you add a card or make some changes, then it start adding those doublons. I don't think I have any other regex that could cause a problem, as the only other one I use is `^###(.+)\n+((?:[^\n#][\n]?)+)`. Any idea how I could solve this? Do you think there is a problem with my regex `^- (.*) \/ (.*)\n`?

**Comment by ddfonseca** [comment]
Man, here works fine again with D /E regex. I think this Mathjax is getting the way because your file has both title and D / E.
Try again just with D / E without Mathjax.

mv obsidian_to_anki_config.ini obsidian_to_anki_config.ini.bkp
python3 obsidian_to_anki.py -c (to reload the config with default values)
Change `Regex = False to Regex = True `on obsidian_to_anki_config.ini file
Run again

**Comment by raphaelhuleux** [comment]
Hey it must have been an error from me for the D / E, I managed to make it work by simply rebooting my computer. 

However, I still can't get the `^- (.*) \/ (.*)\n` regex to work (I also tried `^- (.+) \/ (.+)\n` to avoid mistakes, with only one regex as you advised). The behavior with this regex is strange: it adds cards to Anki but doesn't add IDs to the .md file. Some cards also get duplicate IDs but don't seem to be duplicated in Anki. 
Any idea how to solve hit?

**Comment by ddfonseca** [comment]
> Hey it must have been an error from me for the D / E, I managed to make it work by simply rebooting my computer.
> 
> However, I still can't get the `^- (.*) \/ (.*)\n` regex to work (I also tried `^- (.+) \/ (.+)\n` to avoid mistakes, with only one regex as you advised). The behavior with this regex is strange: it adds cards to Anki but doesn't add IDs to the .md file. Some cards also get duplicate IDs but don't seem to be duplicated in Anki.
> Any idea how to solve hit?

Indeed very strange behavior! I suggest you create another issue, the author of script is off for some reason, but he actually answer very quickly. I'm just a fan trying to help other members, and I don't have a good experience with regex, sorry

**Comment by Pseudonium** [comment]
@ddfonseca Thanks for your help with this - I've been busy with university, so haven't really had time to check this out.

@raphaelhuleux I can't seem to download your zips! Though if you provide an example of the double-id I'll see what I can do about it...

**Comment by Pseudonium** [comment]
Nevermind, I can download your zips now. Though I find this convo a little hard to follow - which file are you having an error in now, and what bug? I have recently updated most regexps with something that should hopefully avoid any 'double id' issues in the future
